cmake/Modules/VolkAddTest.cmake

```diff
@@ -139,17 +139,18 @@ function(VOLK_ADD_TEST test_name executable_name)
     #generate a shell script file that sets the environment and runs the test
     set(sh_file ${CMAKE_CURRENT_BINARY_DIR}/${test_name}_test.sh)
     file(WRITE ${sh_file} "#!${SHELL}\n")
-    if(SHELL_SUPPORTS_IFS)
-      file(APPEND ${sh_file} "export IFS=:\n")
-    else()
-      file(APPEND ${sh_file} "LL=\"$1\" && for tf in \"\$@\"; do LL=\"\${LL}:\${tf}\"; done\n")
-    endif()
-
-    #each line sets an environment variable
-    foreach(environ ${environs})
-      file(APPEND ${sh_file} "export ${environ}\n")
-    endforeach(environ)
-
+    if (NOT CMAKE_CROSSCOMPILING)
+        if(SHELL_SUPPORTS_IFS)
+            file(APPEND ${sh_file} "export IFS=:\n")
+        else()
+            file(APPEND ${sh_file} "LL=\"$1\" && for tf in \"\$@\"; do LL=\"\${LL}:\${tf}\"; done\n")
+        endif()
+
+        #each line sets an environment variable
+        foreach(environ ${environs})
+            file(APPEND ${sh_file} "export ${environ}\n")
+        endforeach(environ)
+    endif(CMAKE_CROSSCOMPILING)
```

Comment:
How is one supposed to use it for cross-compiling?

Reply:
Now *that's* some documentation that I'd read.

Reply:
What I am doing is creating a package that has all the tests in it. So tests run on the target are not running in the build tree.  Are we actually using the bit that runs qemu?

Reply:
We have 2 CI tests that use QEMU. I'd argue that we can consider these tests our cross compile tests.

I'd really love to make cross compiling as simple as possible. Given the fact that I don't have experience with that yet I'd like to understand how to use your changes though. Otherwise I fear things will break again soon.